src/exec/run-batch-submitter.ts

```diff
@@ -164,18 +173,29 @@ export const run = async () => {
         if (pendingTxs > latestTxs) {
           log.info('Detected pending transactions. Clearing all transactions!')
           for (let i = latestTxs; i < pendingTxs; i++) {
-            const response = await sequencerSigner.sendTransaction({
-              to: await sequencerSigner.getAddress(),
-              value: 0,
-              nonce: i,
-            })
-            log.info(`Submitting transaction with nonce: ${i}; hash: ${response.hash}`)
+            const seqSignerAddress = await sequencerSigner.getAddress()
+            const sendFunc = async () => {
+              const tx = sequencerSigner.sendTransaction({
+                to: seqSignerAddress,
+                value: 0,
+                nonce: i,
+              })
+              const response = await tx
+              return response.wait(parseInt(requiredEnvVars.NUM_CONFIRMATIONS, 10))
```

Comment:
Do we want to wait for the transaction inside of the `txFunction` ? Looking at the example here, they only send the tx & don't wait for the receipt - https://github.com/kendricktan/ynatm#quickstart

Reply:
resolved on slack; we are waiting for receipt within the sendTxFunction and setting a long timeout